prow/cmd/webhook-server/clients.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/test-infra/experiment/clustersecretbackup/secretmanager"
+)
+
+const clientListError = "could not make call to list secrets successfully"
+
+type GCPClient struct{
+	client *secretmanager.Client
+}
+
+func newGCPClient(client *secretmanager.Client) *GCPClient {
+	return &GCPClient{
+		client: client,
+	}
+}
+
+func (g *GCPClient) CreateSecret(ctx context.Context, secretID string) error {
+	_, err := g.client.CreateSecret(ctx, secretID)
+	if err != nil {
+		return fmt.Errorf("could not create secret %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) AddSecretVersion(ctx context.Context, secretName string, payload []byte) error {
+	if err := g.client.AddSecretVersion(ctx, secretName, payload); err != nil {
+		return fmt.Errorf("could not add secret data %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) GetSecretValue(ctx context.Context, secretName string, versionName string) ([]byte, error) {
+	exist, err := g.CheckSecret(ctx, secretName)
+    if err != nil {
+        return nil, err
+    }
+    if !exist {
+        return nil, errors.New("secret not exist")
+    }
+	return g.client.GetSecretValue(ctx, secretName, versionName)
+}
+
+func (g *GCPClient) CheckSecret(ctx context.Context, secretName string) (bool, error) {
+	var found bool
```

Comment:
nit. we do not need a bool here. Instead of breaking in the for loop just return true as soon as we find the secret, and return false at the end if we have not found the secret